### ISSUE TITLE
Periodic job transitions to dead on update

### ISSUE THREAD

**Issue by dadgar** [hashicorp/nomad#2268]
Steps to reproduce

1) nomad run example.nomad
2) edit test -> test2
3) nomad run example.nomad
4) nomad status example: It will be dead.

```
job "example" {
  type = "batch"
  datacenters = ["dc1"]
  periodic {
     cron = "* * * * * *"
     prohibit_overlap = true
  }
  group "foo" {
    task "foo" {
      driver = "raw_exec"
      config {
        command = "echo"
        args = ["test2"]
      }
    }
  }
}
```

**Comment by jippi** [comment]
Haha, i've always thought that was the intended state for a periodic job waiting for something to execute... interesting  :D

**Comment by sepulworld** [comment]
Yeah, I was confused with the 'dead' status. Even with the 'dead' status the periodic job still appears to run.

**Comment by briansorahan** [comment]
I wish I would have seen this issue a week ago!
We just started implementing a bulk update to all our periodic jobs and have run into this.

**Comment by briansorahan** [comment]
@dadgar Is this simply a matter of setting status correctly when a periodic job is updated? I'd like to submit a PR if so.
It seems to me like @sepulworld 's comment is correct. The periodic jobs that are marked as "dead" seem to be functioning correctly

**Comment by dadgar** [comment]
@briansorahan Appreciate the offer! I went ahead and fixed it as it is somewhat subtle! In the meantime, the periodic job will have the incorrect state but it will function correctly. We should have a release out soon though!

**Comment by briansorahan** [comment]
@dadgar Thank you!
